src/freenet/client/async/ClientPutter.java

```diff
@@ -287,7 +287,7 @@ public void onSuccess(ClientPutState state, ClientContext context) {
 		}
 		if(super.failedBlocks > 0 || super.fatallyFailedBlocks > 0 || super.successfulBlocks < super.totalBlocks) {
 			// USK auxiliary inserts are allowed to fail.
-			if(!uri.isUSK())
+			if((!uri.isUSK()) && !ctx.getCHKOnly)
```

Comment:
Why the additional parenthesis around `!uri.isUSK()`? They don't seem necessary or helpful in understanding so my inclination is to leave them out.

While this line is changing anyway we might as well fix the whitespace to meet the [coding standards](https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s4.6.2-horizontal-whitespace): this needs a single space between `if` and the following `(`.